packages/mui-material/src/Accordion/Accordion.js

```diff
@@ -91,28 +91,36 @@ const AccordionRoot = styled(Paper, {
       },
     };
   },
-  ({ theme, ownerState }) => ({
-    ...(!ownerState.square && {
-      borderRadius: 0,
-      '&:first-of-type': {
-        borderTopLeftRadius: (theme.vars || theme).shape.borderRadius,
-        borderTopRightRadius: (theme.vars || theme).shape.borderRadius,
-      },
-      '&:last-of-type': {
-        borderBottomLeftRadius: (theme.vars || theme).shape.borderRadius,
-        borderBottomRightRadius: (theme.vars || theme).shape.borderRadius,
-        // Fix a rendering issue on Edge
-        '@supports (-ms-ime-align: auto)': {
-          borderBottomLeftRadius: 0,
-          borderBottomRightRadius: 0,
+  ({ theme }) => ({
+    variants: [
+      {
+        props: (props) => !props.square,
```

Comment:
From https://mui-org.slack.com/archives/C0170JAM7ML/p1708502628309269, I would keep using `ownerState`:
```suggestion
        props: ({ ownerState }) => !ownerState.square,
```

Reply:
No need to, I've changed the API to support this. For more clarity as mentioned in the thread, we don't write:

```
variants: [{
  props: { 'ownerState.square': true }
  style: {}
}]
```

so it may be confusing to ask people to use ownerState only in the callback.